src/Resources/JsonApiResource.php

```diff
@@ -92,284 +85,85 @@ public function toArray($request): array
-    {
-        if (empty($this->resourceRegistrationData['relationships'])) {
-            return;
-        }
-
-        foreach ($this->resourceRegistrationData['relationships'] as $relationKey => $relationData) {
-            $this->mapSingleRelationship($relationKey, $relationData);
-        }
-    }
-
-    /**
-     * Map a single relationship into an included resource
-     */
-    private function mapSingleRelationship($relationKey, $relationData): void
-    {
-        [$resourceData, $resourceClass] = $relationData;
-
-        // The data can be a string reference to relation name on the model or a resource model itself
-        $resourceData = is_string($resourceData)
-            ? $this->convertRelationStringToReference($resourceData)
-            : $resourceData;
-
-        if (! $this->resourceHasData($resourceData, $resourceClass)) {
-            return;
-        }
-
-        // Is this relation a collection or a single resource
-        if (is_subclass_of($resourceClass, JsonApiResourceCollection::class)) {
-            $this->addResourceCollectionRelation($relationKey, $resourceData, $resourceClass);
-
-            return;
-        }
-
-        $this->addResourceRelation($relationKey, $resourceData, $resourceClass);
-    }
-
-    /**
-     * Add a collection relationship to the resource relationships and included data
-     */
-    private function addResourceCollectionRelation(
-        $relationKey,
-        $resourceDataCollection,
-        $resourceCollectionClass
-    ): void {
-        $resourceClass = (new $resourceCollectionClass([]))->collects;
-        $relationshipReferences = [];
-
-        foreach ($resourceDataCollection as $resourceData) {
-            $includedResource = new $resourceClass([$resourceData, $this->maxResourceDepth, $this->resourceDepth + 1]);
-            if (! $includedResource instanceof self) {
-                continue;
-            }
-            $this->addInclude($includedResource);
-            $relationshipReferences[] = $includedResource->toRelationshipReferenceArray();
-        }
-
-        $this->resourceRelationshipReferences[$relationKey] = [
-            'data' => $relationshipReferences,
-        ];
-    }
-
-    /**
-     * Add a relation to the resource
-     */
-    private function addResourceRelation($relationKey, $resourceData, $resourceClass): void
-    {
-        $includedResource = new $resourceClass([$resourceData, $this->maxResourceDepth, $this->resourceDepth + 1]);
-        if (! $includedResource instanceof self) {
-            return;
-        }
-
-        $this->addInclude($includedResource);
-
-        $this->resourceRelationshipReferences[$relationKey] = [
-            'data' => $includedResource->toRelationshipReferenceArray(),
-        ];
-    }
-
-    /**
-     * Check if the resource object has data
+     * @param  Request  $request
      */
-    private function resourceHasData($resourceData, $resourceClass): bool
+    public function with($request): array
```

Comment:
Should this be moved to relations trait? Or is it part of the JsonResource from Laravel?

Reply:
Part of JsonResource from Laravel. Leaving it for now.